### ISSUE TITLE
Trim fields

### ISSUE THREAD

**Issue by alehandrof** [statamic/v2-hub#2133]
Provide an option to trim fields saved by the built-in forms or the Workshop add-on.

**Comment by lukeagsmith** [comment]
I'd also love for CP fields to get trimmed. Clients don't even know how to copy-paste properly from Word!

**Comment by shabbarabbas** [comment]
I agree , i am in situation where email verification fail if there is a space after it , and it is hard to judge the space with eye , is there any solution for this ??
![blank space](https://user-images.githubusercontent.com/6230508/45359599-7253cc00-b5e6-11e8-90c6-60675d02d5d8.png)

**Comment by shabbarabbas** [comment]
anybody solve the issue please share here

**Comment by lukeagsmith** [comment]
@shabbarabbas If this is your own form and not in the CP, then just trim it with Javascript on submission of the form.

**Comment by shabbarabbas** [comment]
on the front-end i have added a jquery code 
`	$(".trim-field").blur(function() {
	 var str = $(this).val();
	 		str = str.trim();
	    $(this).val(str);
	});
`
and in the template add **trim-field**  class to the field that need trim , it will remove trailing and leading spaces as soon you click out of the field 

any way this workaround is for the front-end where you can add custom class to the field 
anyone can submit CP workaround 
Hopefully Statamic Team will solve the issue

**Comment by JesseLeite** [comment]
Should be fixed in next version 👍 

![](https://memegenerator.net/img/instances/26042452/fresh-trim.jpg)
